R/pass_fail_condi.R

```diff
@@ -67,6 +67,16 @@ evaluate_condition <- function(condi, grader_args, learnr_args) {
            "value" = evaluate_condi_value(condi$x, learnr_args$last_value)
          )
 
+  # if we compare something like a vector or dataframes to one another
+  # we need to collapse the result down to a single boolean value
+  if (length(res) > 1) {
```

Comment:
Maybe also check for size = 0?

```r
if (length(res) == 0) {
  warning(glue::glue("I got a length of {length(res)}, instead of 1 during the conditional check.\nDid you use == ? If so, consider using idential()")) # nolint
  res <- FALSE
}
```

Reply:
the check for size = 0 is already made in the checkmate call

https://github.com/rstudio-education/grader/blob/d678f79e488dafc12781b59a0055e3f3c96b3b4b/R/pass_fail_condi.R#L85